components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/LoginDialogFragment.kt

```diff
@@ -179,19 +179,8 @@ internal class LoginDialogFragment : PromptDialogFragment() {
             override fun afterTextChanged(editable: Editable) {
                 // Note that password is accessed by `fun update`
                 password = editable.toString()
-
-                if (password.isEmpty()) {
-                    setViewState(
-                        confirmButtonEnabled = false,
-                        passwordErrorText =
-                        context?.getString(R.string.mozac_feature_prompt_error_empty_password)
-                    )
-                } else {
-                    setViewState(
-                        confirmButtonEnabled = true,
-                        passwordErrorText = ""
-                    )
-                }
+                // Update accesses member state, so it must be called after password is set
+                update()
```

Comment:
A password change can't cause a record to become a dupe (or stop being one). I don't think you need to query the store in this case at all.